src/box/memtx_tx.c

```diff
@@ -349,17 +349,25 @@ void
 memtx_tx_handle_conflict(struct txn *breaker, struct txn *victim)
 {
 	assert(breaker->psn != 0);
-	if (victim->status != TXN_INPROGRESS) {
+	if (victim->status != TXN_INPROGRESS &&
+	    victim->status != TXN_IN_READ_VIEW) {
 		/* Was conflicted by somebody else. */
 		return;
 	}
 	if (stailq_empty(&victim->stmts)) {
-		/* Send to read view. */
-		victim->status = TXN_IN_READ_VIEW;
-		victim->rv_psn = breaker->psn;
-		rlist_add_tail(&txm.read_view_txs, &victim->in_read_view_txs);
+		/* Send to read view, perhaps a deeper one. */
+		if (victim->status != TXN_IN_READ_VIEW) {
+			victim->status = TXN_IN_READ_VIEW;
+			victim->rv_psn = breaker->psn;
+			rlist_add_tail(&txm.read_view_txs,
+				       &victim->in_read_view_txs);
+		} else {
+			victim->rv_psn = MIN(victim->rv_psn, breaker->psn);
+		}
 	} else {
 		/* Mark as conflicted. */
+		if (victim->status == TXN_IN_READ_VIEW)
+			rlist_del(&victim->in_read_view_txs);
```

Comment:
What is the point of this snippet?

Reply:
A transaction is on read view list when and only when the status is TXN_IN_READ_VIEW. If we clear the status - we must remove from list.